docs/OV_Runtime_UG/supported_plugins/GNA.md

```diff
@@ -10,371 +9,131 @@ to save power and free CPU resources.
+For details, see description of the `ov::intel_gna::execution_mode` property.
 
-## <a name="models-and-layers-limitations">Models and Layers Limitations</a>
+## Recovery from Interruption by High-Priority Windows Audio Processes\*
 
-Because of specifics of hardware architecture, Intel® GNA supports a limited set of layers, their kinds and combinations.
-For example, you should not expect the GNA Plugin to be able to run computer vision models, except those specifically adapted for the GNA Plugin, because the plugin does not fully support 2D convolutions.
+GNA is designed for real-time workloads such as noise reduction.
+For such workloads, processing should be time constrained, otherwise extra delays may cause undesired effects such as
+*audio glitches*. To make sure that processing can satisfy real-time requirements, the GNA driver provides a Quality of Service
+(QoS) mechanism, which interrupts requests that might cause high-priority Windows audio processes to miss
+the schedule, thereby causing long running GNA tasks to terminate early.
 
-For the list of supported layers, see the **GNA** column of the **Supported Layers** section in [Supported Devices](Supported_Devices.md).
+Applications should be prepared for this situation. We recommend to use Automatic QoS Feature described below.
 
-Limitations include:
+### Automatic QoS Feature on Windows*
 
-- Only 1D convolutions are natively supported on the HW prior to GNA 3.0; 2D convolutions have specific limitations (see the table above).
-- The number of output channels for convolutions must be a multiple of 4.
-- The maximum number of filters is 65532 for GNA 2.0 and 8192 for GNA 3.0.
-- Transpose layer support is limited to the cases where no data reordering is needed or when reordering is happening for two dimensions, at least one of which is not greater than 8.
-- Splits and concatenations are supported for continuous portions of memory (e.g., split of 1,2,3,4 to 1,1,3,4 and 1,1,3,4 or concats of 1,2,3,4 and 1,2,3,5 to 2,2,3,4).
-- For Multiply, Add and Subtract layers, auto broadcasting is only supported for constant inputs.
+Starting with 2021.4.1 release of OpenVINO and 03.00.00.1363 version of Windows* GNA driver, a new execution mode `ov::intel_gna::ExecutionMode::HW_WITH_SW_FBACK` is introduced
+to assure that workloads satisfy real-time execution. In this mode, the GNA driver automatically falls back on CPU for a particular infer request
+if the HW queue is not empty, so there is no need for explicitly switching between GNA and CPU.
 
-### Support for 2D Convolutions in Previous Generations of GNA Hardware
+@sphinxdirective
+.. tab:: C++
+
+   .. code-block:: cpp
 
-The Intel® GNA 1.0 and 2.0 hardware natively supports only 1D convolutions.
+      ov::Core core;
+      model = core.read_model(path_to_model);
+      compiled_model = core.compile_model(model, "GNA", ov::intel_gna::execution_mode(ov::intel_gna::ExecutionMode::HW_WITH_SW_FBACK));
```

Comment:
1. We put code snippets to docs/snippets to make sure they are compilable 
2. Can you shorten the line to avoid horizontal scroll bars after docs compilation?

Reply:
Done